src/worker.cpp

```diff
@@ -1185,4 +1183,305 @@ void Worker::collectFrusFromJson()
+{
+    logging::logMessage("BackupAndRestore is started.");
+    const std::string& l_primaryHwPath = getBackupHwPath("primary");
+    const std::string& l_secondaryHwPath = getBackupHwPath("secondary");
+
+    if (l_primaryHwPath.empty() && l_secondaryHwPath.empty())
+    {
+        logging::logMessage(
+            "Both primary and secondary path's are empty, can't initiate BackupAndRestore.");
+        return;
+    }
+
+    const auto& l_backupMap = getBackupMap();
+    if (std::holds_alternative<std::monostate>(l_backupMap))
+    {
+        logging::logMessage(
+            "Couldn't find system specific backup map, can't initiate BackupAndRestore.");
+        return;
+    }
+
+    types::VPDMapVariant l_parsedPrimaryVpdMap;
+    if (!l_primaryHwPath.empty())
+    {
+        fillVPDMap(l_primaryHwPath, l_parsedPrimaryVpdMap);
+    }
+
+    types::VPDMapVariant l_parsedSecVpdMap;
+    if (!l_secondaryHwPath.empty())
+    {
+        fillVPDMap(l_secondaryHwPath, l_parsedSecVpdMap);
+    }
+
+    if (!l_primaryHwPath.empty() && !l_secondaryHwPath.empty() &&
+        l_parsedPrimaryVpdMap.index() != l_parsedSecVpdMap.index())
+    {
+        logging::logMessage(
+            "Primary and secondary VPD types are not matching.");
+        return;
+    }
+
+    // Implement backup and restore for IPZ type VPD
+    if (auto l_backupMapPtr = std::get_if<types::IpzBackupMap>(&l_backupMap))
+    {
+        types::IPZVpdMap* l_parsedVpdPtr = nullptr;
+        if (!(l_parsedVpdPtr = std::get_if<types::IPZVpdMap>(&io_parsedVpdMap)))
+        {
+            if (std::holds_alternative<std::monostate>(io_parsedVpdMap))
+            {
+                l_parsedVpdPtr = new types::IPZVpdMap();
+            }
+            else
+            {
+                logging::logMessage("Parsed VPD is not of IPZ type.");
+                return;
+            }
+        }
+
+        types::IPZVpdMap* l_primaryVpdPtr;
+        if (!(l_primaryVpdPtr =
+                  std::get_if<types::IPZVpdMap>(&l_parsedPrimaryVpdMap)))
+        {
+            if (std::holds_alternative<std::monostate>(l_parsedPrimaryVpdMap))
+                l_primaryVpdPtr = new types::IPZVpdMap();
+            else
+            {
+                logging::logMessage("Primary VPD is not of IPZ type.");
+                return;
+            }
+        }
+
+        types::IPZVpdMap* l_secVpdPtr;
+        if (!(l_secVpdPtr = std::get_if<types::IPZVpdMap>(&l_parsedSecVpdMap)))
+        {
+            if (std::holds_alternative<std::monostate>(l_parsedSecVpdMap))
+                l_secVpdPtr = new types::IPZVpdMap();
+            else
+            {
+                logging::logMessage("Secondary VPD is not of IPZ type.");
+                return;
+            }
+        }
+
+        backupAndRestoreIpzVpd(*l_parsedVpdPtr, *l_backupMapPtr,
+                               *l_primaryVpdPtr, *l_secVpdPtr);
+    }
+    else if (std::holds_alternative<types::KeywordVpdMap>(
```

Comment:
No need of this else condition. Just have a note mentioned there to support any other type.